kube-core/src/response.rs

```diff
@@ -35,8 +35,53 @@ pub struct Status {
     pub details: Option<StatusDetails>,
 }
 
+impl Status {
+    /// Returns a successful `Status`
+    pub fn success() -> Self {
+        Status {
+            status: Some(StatusSummary::Success),
+            code: 0,
+            message: String::new(),
+            reason: String::new(),
+            details: None,
+        }
+    }
+
+    /// Returns an unsuccessful `Status`
+    pub fn failure(message: &str, reason: &str) -> Self {
+        Status {
+            status: Some(StatusSummary::Failure),
+            code: 0,
+            message: message.to_string(),
+            reason: reason.to_string(),
+            details: None,
+        }
+    }
+
+    /// Sets an explicit HTTP status code
+    pub fn with_code(mut self, code: u16) -> Self {
+        self.code = code;
+        self
+    }
+
+    /// Adds details to the `Status`
+    pub fn with_details(mut self, details: StatusDetails) -> Self {
+        self.details = Some(details);
+        self
+    }
+}
```

Comment:
can we also add:

```
pub fn is_failure(&self) -> bool;
pub fn is_success(&self) -> bool;
```

while we are here? :-)

Reply:
Done